src/Simplex_tree/include/gudhi/Simplex_tree.h

```diff
@@ -1767,6 +2115,88 @@ class Simplex_tree {
     return sh; // None of its faces has the same filtration.
   }
 
+ public:
+  // intrusive list of Nodes with same label using the hooks
+  typedef boost::intrusive::member_hook<Hooks_simplex_base_link_nodes,
+                                        typename Hooks_simplex_base_link_nodes::Member_hook_t,
+                                        &Hooks_simplex_base_link_nodes::list_max_vertex_hook_>
+      List_member_hook_t;
+  // auto_unlink in Member_hook_t is incompatible with constant time size
+  typedef boost::intrusive::list<Hooks_simplex_base_link_nodes, List_member_hook_t,
+                                 boost::intrusive::constant_time_size<false>>
+      List_max_vertex;
+  // type of hooks stored in each Node, Node inherits from Hooks_simplex_base
+  typedef typename std::conditional<Options::link_nodes_by_label, Hooks_simplex_base_link_nodes,
+                                    Hooks_simplex_base_dummy>::type Hooks_simplex_base;
+  /** Data structure to access all Nodes with a given label u. Can be used for faster
+   * computation. */
+ private:
+  // if Options::link_nodes_by_label is true, store the lists of Nodes with
+  // same label
+  typedef typename std::conditional<Options::link_nodes_by_label, Nodes_by_label_intrusive_list<Simplex_tree>,
+                                    Nodes_by_label_dummy<Simplex_tree>>::type Nodes_by_label_data_structure;
+
+  /** Only if Options::link_nodes_by_label is true, nodes_with_label_[u] returns a
+   * range of all Nodes in the Simplex_tree with the label u.*/
+  Nodes_by_label_data_structure nodes_by_label_;
+
+ public:
+  List_max_vertex* nodes_by_label(Vertex_handle u) {
+    if constexpr (Options::link_nodes_by_label) {
+      return nodes_by_label_.find(u);
+    }
+    return nullptr;
+  }
+
+  // basic methods implemented for Nodes, and not Simplex_handle. The hooks in
+  // nodes_by_label_ gives access to Nodes.
+ public:
+  // set of methods taking Node as input. For internal use only.
+  /** Returns the Siblings containing a simplex.*/
+  static Siblings* self_siblings(Node& node, Vertex_handle v) {
+    if (node.children()->parent() == v) {
+      return node.children()->oncles();
+    } else {
+      return node.children();
+    }
+  }
+
+  int dimension(Node& node, Vertex_handle u) {
+    Siblings* curr_sib = self_siblings(node, u);
+    int dim = 0;
+    while (curr_sib != nullptr) {
+      ++dim;
+      curr_sib = curr_sib->oncles();
+    }
+    return dim - 1;
+  }
+  /* \brief Returns true if the node in the simplex tree pointed by
+   * sh has children. node must have label u*/
```

Comment:
What is sh?

Reply:
This method/doc is not from me. But I still replaced "by sh" with "by the given simplex handle".